src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -470,8 +470,18 @@ protected String remoteCheckpointDocID() {
 
     public Set<String> getPendingDocumentIDs() {
         synchronized (lockPendingDocIDs) {
-            if (isPull() || (isRunning() && pendingDocIDs != null))
-                return pendingDocIDs;
+
+            /** 
+             * Bug causing code                
+             * Expected Functionality:
+             *      While offline, I expect the return value of this method to reflect
+             *      the number of pendingDocIDs inline with database mutations 
+             * Actual Behavior:
+             *      The commented out lines below, output a cached value for pendingDocIDs 
+             *
+             * if (isPull() || (isRunning() && pendingDocIDs != null))
+             *   return pendingDocIDs;
+            */
```

Comment:
Removing this check entirely is not correct. First this method should return null for pull replicator.

It's best to port the changes from CBL iOS: 

https://github.com/couchbase/couchbase-lite-ios/blob/master/Source/API/CBLReplication.m#L445-L451